staging/src/k8s.io/apiextensions-apiserver/pkg/controller/openapi/builder/builder.go

```diff
@@ -74,6 +87,9 @@ type Options struct {
 	// Convert to OpenAPI v2.
 	V2 bool
 
+	// Used only by Server Side Apply. Create a V2 schema that includes fields from OpenAPI V3 for SSA to make better decisions
```

Comment:
It's unclear what these better decisions are ;-)

Reply:
this needs much more explaination, especially because it leaks into other packages. What does this do *exactly*?

Reply:
The V2 flag is a parameter to feed the OpenAPI spec through the [ToStructuralOpenAPIV2](https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/apiextensions-apiserver/pkg/controller/openapi/v2/conversion.go#L25) flag. 

It does three things: StripValueValidation, StripNullable, and unsets a field if XPreserveUnknownFields is true. I believe this last criteria is not strictly related to v2 but is [required by kubectl](https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/apiextensions-apiserver/pkg/controller/openapi/v2/conversion.go#L76-L89)

Server side apply sets V2: false originally but that is a bit inaccurate because the output is still an OpenAPI v2 spec. Logically speaking V2: false should represent a v3 spec. SSA only set the flag to false because it doesn't want to go through the ToStructuralOpenAPIV2, specifically XPreserveUnknownFields:true fields to be nulled.

I think the ideal way to do this is to create a flag like StripTypesWithPreserveUnknownFields and completely delete the V2 flag (because it can be expressed as a combination of this flag, StripValueValidation, and StripNullable) but that creates a much bigger delta in the code and is more error prone.

I updated the flag to `OverrideToStructuralOpenAPIV2` and added a brief description of its purpose in the code. Let me know your thoughts, thanks!